languages/js/src/types.ts

```diff
@@ -591,17 +590,32 @@ export interface Options {
  *
  * @internal
  */
-export function isIterableIterator(x: any): boolean {
-  return typeof x?.next === 'function' && Symbol.iterator in Object(x);
+export function isIterableIterator(x: unknown): x is IterableIterator<unknown> {
```

Comment:
so this `x is y` notation means that it adds a type constraint in the calling scope depending on if it returns true or false?

Reply:
yep: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates